proposal for Nixpkgs documentation (#2)
nix-community/projects: proposals/stdenv.md

This is testimony that users are invested in the ecosystem, care for better documentation, and trust maintainers to execute on their plans.

While the mentioned chapters in the Nixpkgs manual are central to the entire ecosystem and critically important early on in user onboarding, they are by far the weakest parts of documentation and a clear hindrance to adoption – well known among new users, contributors, and maintainers for lack of crucial information, poor discoverability, and densely written text full of unfamiliar concepts and terminology (to the point of being the subject of [elaborate mockery](https://ianthehenry.com/posts/how-to-learn-nix/)).
Most importantly, this part of documentation is **effectively unmaintained** as currently no one has the resources to take ownership due to its enormous size.

Nixpkgs is a large software project, and one of the largest public GitHub repositories.

> The code itself originated from research projects in [2004](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwi5o8mAwOn_AhWBS_EDHRolCnQQFnoECA4QAQ&url=https%3A%2F%2Fedolstra.github.io%2Fpubs%2Fnspfssd-lisa2004-final.pdf&usg=AOvVaw10O4UewT2sh6kUSXYR3jNC&opi=89978449) and [2006](https://nixos.org/docs/SCR-2005-091.pdf), and albeit always serving its purpose, for various reasons has grown without clear direction since and saw, at best, sparse efforts to provide approachable documentation.

Reviewer: Please use direct link instead of a Google redirect.

```suggestion
The code itself originated from research projects in [2004](https://edolstra.github.io/pubs/nspfssd-lisa2004-final.pdf) and [2006](https://nixos.org/docs/SCR-2005-091.pdf), and albeit always serving its purpose, for various reasons has grown without clear direction since and saw, at best, sparse efforts to provide approachable documentation.
```

## How will you accomplish the work? Please provide a list of deliverables with associated effort and cost of each deliverable.

Goals: Identify changes that have lasting effect after the project is finished.

#### Technical

- Split the manual into separate pages (2w, 8 000 EUR)
- Automatically collect and add SEO-relevant metadata to manual pages (1w, 4 000 EUR)
- Display full table of contents in one place (2d, 1 600 EUR)
- Changes to phases/hooks/shell-functions documentation should not require a mass rebuild (1w, 4 000 EUR)
- Render documentation for shell utilities from source (3d, 2 400 EUR)
- Allow for multiple examples in in-code comments (1w, 4 000 EUR)
- Stable URLs to all manual versions (1w, 4 000 EUR)
- "Edit" button on each page (2d, 1 600 EUR)
- Add breadcrumb navigation (1d, 800 EUR)
- Display maintainers and last-edited date for each page (2d, 1 600 EUR)
- Provide infrastructure for auto-testing code samples (5w, 20 000 EUR)
- Add markup support for glossary terms (1d, 800 EUR)

> - Preprocess manual output for styling and design new style sheet (3w, 12 000 EUR)

Reviewer 2: ```suggestion
- Live-preview of changes (2d, 1 600 EUR)
```

> 

Reviewer 2: ```suggestion
- Preview deployments of pull requests (1d, 800 EUR)
- Live-preview of changes (2d, 1 600 EUR)

```
